src/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/nonzero.hpp

```diff
@@ -19,18 +19,22 @@ namespace LayerTestsDefinitions {
 using ConfigMap = typename std::map<std::string, std::string>;
 
 using NonZeroLayerTestParamsSet = typename std::tuple<
-    InferenceEngine::SizeVector,          // Input shapes
+    ov::test::InputShape,                 // Input tensors shape
     InferenceEngine::Precision,           // Input precision
-    LayerTestsUtils::TargetDevice,        // Device name
+    std::string,                          // Device name
     ConfigMap>;                           // Additional network configuration
 
 class NonZeroLayerTest : public testing::WithParamInterface<NonZeroLayerTestParamsSet>,
-                         virtual public LayerTestsUtils::LayerTestsCommon {
+                         virtual public ov::test::SubgraphBaseTest {
 public:
     static std::string getTestCaseName(const testing::TestParamInfo<NonZeroLayerTestParamsSet>& obj);
 
 protected:
     void SetUp() override;
+    void generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) override;
+
+protected:
+    size_t startFrom = 0, range = 10;
```

Comment:
minor: maybe we can set range with smaller value like 2 or 5, to have more zeros in the input tensor to have a diverse distribution. In my understanding, in real models, there should be lot of zeros in input tensor.

Reply:
Done (changed to 3)